### PR TITLE
Re-use transceiver (via ReplaceTrack) if a down track is going to be resumed.

### PR DESCRIPTION
**Pull request by boks1971**
- Cache transceiver when closing a down track if that down track is expected to be resumed
- When creating a new down track, check for cached transceiver that can be re-used.
- If possible, re-use.
- If not, stop the transceiver to ensure that there is no transceiver in active state if not used.

### CODE REVIEW

File: pkg/rtc/wrappedreceiver.go [hunk 1]
```diff
@@ -67,8 +67,15 @@ type DummyReceiver struct {
...
-	downtracks       map[livekit.ParticipantID]sfu.TrackSender
+
+	downtrackLock sync.Mutex
+	downtracks    map[livekit.ParticipantID]sfu.TrackSender
+
+	settingsLock          sync.Mutex
+	maxExpectedLayerValid bool
+	maxExpectedLayer      int32
+	pausedValid           bool
+	paused                bool
```
Comment (boks1971) [inline pkg/rtc/wrappedreceiver.go:78]: Adding these as I think these will be need to be set up properly once the receiver is determined.

File: pkg/rtc/mediatracksubscriptions.go [hunk 2]
```diff
@@ -272,7 +310,18 @@ func (t *MediaTrackSubscriptions) RemoveSubscriber(participantID livekit.Partici
 	t.subscribedTracksMu.Unlock()
 
 	if subTrack != nil {
-		subTrack.DownTrack().CloseWithFlush(!willBeResumed)
+		dt := subTrack.DownTrack()
```
Comment (cnderrauber) [inline pkg/rtc/mediatracksubscriptions.go:313]: can we use a function for this? duplicate code in RemoveAllSubscribers

**Review by cnderrauber — APPROVED** [review]
Great work!!

File: pkg/rtc/mediatracksubscriptions.go (hunk 2 above)
Comment (boks1971) [reply pkg/rtc/mediatracksubscriptions.go]: Thank you @cnderrauber . Added this commit to address this and also more defensive in uncaching the transceiver - https://github.com/livekit/livekit/pull/785/commits/3c0cc04c0405ba042bb370530fa42eb9c42d7113

**Pull request merged**